be/src/formats/json/nullable_column.cpp

```diff
@@ -0,0 +1,183 @@
+
+static Status add_nullable_binary_column(Column* column, const TypeDescriptor& type_desc, const std::string& name,
+                                         simdjson::ondemand::value* value) {
+    auto nullable_column = down_cast<NullableColumn*>(column);
+
+    auto& null_column = nullable_column->null_column();
+    auto& data_column = nullable_column->data_column();
+
+    try {
+        if (value->is_null()) {
+            data_column->append_default(1);
+            null_column->append(1);
+            return Status::OK();
+        }
+
+        RETURN_IF_ERROR(add_binary_column(data_column.get(), type_desc, name, value));
+
+        null_column->append(0);
+        return Status::OK();
+    } catch (simdjson::simdjson_error& e) {
+        auto err_msg = strings::Substitute("Failed to parse value as binary type, column=$0, error=$1", name,
+                                           simdjson::error_message(e.error()));
+        return Status::DataQualityError(err_msg);
+    }
+}
+
+static Status add_nullable_boolean_column(Column* column, const TypeDescriptor& type_desc, const std::string& name,
+                                          simdjson::ondemand::value* value) {
+    auto nullable_column = down_cast<NullableColumn*>(column);
+
+    auto& null_column = nullable_column->null_column();
+    auto& data_column = nullable_column->data_column();
+
+    try {
+        if (value->is_null()) {
+            data_column->append_default(1);
+            null_column->append(1);
+            return Status::OK();
+        }
+
+        RETURN_IF_ERROR(add_boolean_column(data_column.get(), type_desc, name, value));
+
+        null_column->append(0);
+        return Status::OK();
+    } catch (simdjson::simdjson_error& e) {
+        auto err_msg = strings::Substitute("Failed to parse value as boolean, column=$0, error=$1", name,
+                                           simdjson::error_message(e.error()));
+        return Status::DataQualityError(err_msg);
+    }
+}
+
+static Status add_nullable_column(Column* column, const TypeDescriptor& type_desc, const std::string& name,
+                                  simdjson::ondemand::value* value) {
+    // The type mappint should be in accord with JsonScanner::_construct_json_types();
+    switch (type_desc.type) {
+    case TYPE_BIGINT:
+        return add_nullable_numeric_column<int64_t>(column, type_desc, name, value);
+    case TYPE_INT:
+        return add_nullable_numeric_column<int32_t>(column, type_desc, name, value);
+    case TYPE_SMALLINT:
+        return add_nullable_numeric_column<int16_t>(column, type_desc, name, value);
+    case TYPE_TINYINT:
+        return add_nullable_numeric_column<int8_t>(column, type_desc, name, value);
+    case TYPE_DOUBLE:
+        return add_nullable_numeric_column<double>(column, type_desc, name, value);
+    case TYPE_FLOAT:
+        return add_nullable_numeric_column<float>(column, type_desc, name, value);
+    case TYPE_BOOLEAN:
```

Comment:
```suggestion
    case TYPE_VARCHAR
```

Reply:
The `BOOLEAN` should be treated as `VARCHAR`, which is the default switch case.